frontend/src/components/TreeViewer/TreeViewer.jsx

```diff
@@ -153,7 +160,7 @@ const setOpacity = (data, link, currentOpacity) => {
     currentOpacity !== VConf.FADE_OPACITY &&
     currentOpacity !== VConf.FULL_OPACITY
   ) {
-    console.log("WARNING: invalid passed in opacity...");
+    log("setOpacity: invalid passed in opacity...", Logger.LEVEL_WARNING);
```

Comment:
I think it might be  a better idea to have the logger workflow like so:
1. Let the file that's using the logger set the minimum level of logging
2.  Have functions that define the log level, do the logging. For example:
instead of `log("something", Logger.LEVEL_WARNING)`
lets have something like `log.warn("something")`, and have that prompt show up like "Warning: something" where warning is colored orange. Similarly, error would be red, info would be green, debug would be blue (or a color scheme like that) 

I feel that a strategy like that might make the logger a bit more intuitive to use? what do you think

Reply:
Can we have coloured stuff in the console? If so then it makes a ton of sense to have different logging functions; if not it seems that it would cause unnecessary code duplication. 
Also why would we want a particular file to log at one level whereas another file logging at another level? Seems that it would make it more difficult to debug, since many actions taken by the user go through multiple files

Reply:
Yea, I like the idea of log.warn, log.info, etc. And yes, we can have different colours in the console: https://stackoverflow.com/questions/7505623/colors-in-javascript-console

Reply:
ok, fixed, please have a look guys